src/Symfony/Component/Serializer/Encoder/JsonDecode.php

```diff
@@ -95,8 +95,8 @@ public function decode($data, $format, array $context = array())
             $decodedData = json_decode($data, $associative, $recursionDepth);
         }
 
-        if (JSON_ERROR_NONE !== $this->lastError = json_last_error()) {
-            throw new UnexpectedValueException(JsonEncoder::getLastErrorMessage());
+        if (JSON_ERROR_NONE !== $this->lastError = json_last_error() and false === $decodedData) {
```

Comment:
Please use `&&` instead of `and` (it doesn't have the priority).

Reply:
In fact using && will break precedence, but it should still be used, so that both operands should be swapped

Reply:
Ok, this is done.

Reply:
done